app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt

```diff
@@ -203,10 +213,66 @@ class MainActivity :
             }
         }
     }
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted: Boolean ->
+        if (isGranted) {
+            //sendNotification(this)
+
+            // Permission is granted. Continue the action or workflow in your
+            // app.
+        } else {
+            // Explain to the user that the feature is unavailable because the
+            // features requires a permission that the user has denied. At the
+            // same time, respect the user's decision. Don't link to system
+            // settings in an effort to convince the user to change their
+            // decision.
+        }
+    }
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         Timber.d("Main Activity onCreate")
         super.onCreate(savedInstanceState)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        when {
+            ContextCompat.checkSelfPermission(
+                this, Manifest.permission.POST_NOTIFICATIONS
+            ) == PackageManager.PERMISSION_GRANTED -> {
+                // You can use the API that requires the permission.
+                Timber.tag("NOTIFICATION").e("onCreate: PERMISSION GRANTED")
+                //sendNotification(this)
+            }
+            shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS) -> {
+                Snackbar.make(
+                    findViewById(R.id.root),
```

Comment:
Trying to show a snackbar before the content is set crashes the application. 

```
Caused by: java.lang.IllegalArgumentException: No suitable parent found from the given view. Please provide a valid view.
                            	at com.google.android.material.snackbar.Snackbar.makeInternal(Snackbar.java:198)
                            	at com.google.android.material.snackbar.Snackbar.make(Snackbar.java:157)
                            	at [au.com](http://au.com/).shiftyjelly.pocketcasts.ui.MainActivity.onCreate(MainActivity.kt:247)
```

Let us move the surrounding if-block below `setContentView(view)` and extract it into a separate method.

Reply:
I moved the code in a seperate function, as suggested. The function will be executed after setContentView